File: algorithms/[896]单调数列/solution.ts
// topics = ["数组"]

/**
 * Iterate
 * time O(n), space O(1), n 为数组长度
 */
function isMonotonic(A: number[]): boolean {
  const n = A.length;
  let inc = true;
  let dec = true;

  for (let i = 0; i < n - 1; i += 1) {
    if (A[i] > A[i + 1]) {
      inc = false;
    }
    if (A[i] < A[i + 1]) {
      dec = false;
    }
  }

  return inc || dec;
}

File: algorithms/[896]单调数列/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        n = len(A)
        diff = A[-1] - A[0]

        for i in range(1, n):
            if diff == 0 and A[i] != A[i - 1]:
                return False
            if diff * (A[i] - A[i - 1]) < 0:
                return False

        return True
